CMake generation fails on Windows when path contains spaces

### Problem

CMake generation fails due to unescaped include argument to MSVC:
```
[137/144] Performing configure step for 'lpeg'
loading initial cache file D:/evil path/neovim/cmake-build-debug/.deps/build/tmp/lpeg-cache-Debug.cmake
-- The C compiler identification is MSVC 19.43.34808.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.43.34808/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.5s)
-- Generating done (0.0s)
-- Build files have been written to: D:/evil path/neovim/cmake-build-debug/.deps/build/src/lpeg-build
[138/144] Performing build step for 'lpeg'
FAILED: build/src/lpeg-stamp/lpeg-build D:/evil path/neovim/cmake-build-debug/.deps/build/src/lpeg-stamp/lpeg-build 
C:\Windows\system32\cmd.exe /C "cd /D "D:\evil path\neovim\cmake-build-debug\.deps\build\src\lpeg-build" && "C:\Program Files\CMake\bin\cmake.exe" --build . && "C:\Program Files\CMake\bin\cmake.exe" -E touch "D:/evil path/neovim/cmake-build-debug/.deps/build/src/lpeg-stamp/lpeg-build""
[1/7] Building C object CMakeFiles\lpeg.dir\lpcap.c.obj
FAILED: CMakeFiles/lpeg.dir/lpcap.c.obj 
"C:\Program Files\CMake\bin\cmake.exe" -E env CCACHE_SLOPPINESS=pch_defines,time_macros C:/Lang/Strawberry/c/bin/ccache.exe C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1443~1.348\bin\Hostx64\x64\cl.exe  /nologo   -ID:/evil path/neovim/cmake-build-debug/.deps/usr/include -ID:/evil path/neovim/cmake-build-debug/.deps/usr/include/luajit-2.1 /Zi /Ob0 /Od /RTC1 -MDd -w /showIncludes /FoCMakeFiles\lpeg.dir\lpcap.c.obj /FdCMakeFiles\lpeg.dir\lpeg.pdb /FS -c "D:\evil path\neovim\cmake-build-debug\.deps\build\src\lpeg\lpcap.c"
cl : Command line warning D9024 : unrecognized source file type 'path/neovim/cmake-build-debug/.deps/usr/include', object file assumed
cl : Command line warning D9027 : source file 'path/neovim/cmake-build-debug/.deps/usr/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'path/neovim/cmake-build-debug/.deps/usr/include/luajit-2.1', object file assumed
cl : Command line warning D9027 : source file 'path/neovim/cmake-build-debug/.deps/usr/include/luajit-2.1' ignored
D:\evil path\neovim\cmake-build-debug\.deps\build\src\lpeg\lpcap.c(2): fatal error C1083: Cannot open include file: 'lua.h': No such file or directory
```

Part of CMakeCache for lpeg build:
```
//Flags used by the C compiler during all build types.
CMAKE_C_FLAGS:STRING=-ID:/evil path/neovim/cmake-build-debug/.deps/usr/include -ID:/evil path/neovim/cmake-build-debug/.deps/usr/include/luajit-2.1
```

Culprit seems to be this:
https://github.com/neovim/neovim/blob/614c9322d50052c76fb3e6e1be7536a972ff0902/cmake.deps/CMakeLists.txt#L76
Escaping those results in successful generation.

A quick project wide search after `-I$` finds 4 locations. And that's only `-I`.
I would guess that in most scenarios, CMake provides appropiate higher level mechanisms like `include_directories`.

### Steps to reproduce

Clone into directory containing a space
Do CMake generation that uses MSVC
-> Generation fails

### Expected behavior

CMake generation should succeed even if the repo path contains spaces

### Nvim version (nvim -v)

master

### Vim (not Nvim) behaves the same?

-

### Operating system/version

Windows 11

### Terminal name/version

-

### $TERM environment variable

-

### Installation

-

--- a/cmake.deps/CMakeLists.txt
+++ b/cmake.deps/CMakeLists.txt
@@ -73,7 +73,7 @@ if(HAS_OG_FLAG)
   set(DEFAULT_MAKE_CFLAGS CFLAGS+=-Og ${DEFAULT_MAKE_CFLAGS})
 endif()
 
-set(DEPS_INCLUDE_FLAGS "-I${DEPS_INSTALL_DIR}/include -I${DEPS_INSTALL_DIR}/include/luajit-2.1")
+set(DEPS_INCLUDE_FLAGS "-I\"${DEPS_INSTALL_DIR}/include\" -I\"${DEPS_INSTALL_DIR}/include/luajit-2.1\"")
 
 # If the macOS deployment target is not set manually (via $MACOSX_DEPLOYMENT_TARGET),
 # fall back to local system version. Needs to be done here and in top-level CMakeLists.txt.
@@ -96,10 +96,10 @@ else()
   find_package(Lua 5.1 EXACT)
   if(LUAJIT_FOUND)
     set(LUA_ENGINE LuaJit)
-    string(APPEND DEPS_INCLUDE_FLAGS " -I${LUAJIT_INCLUDE_DIR}")
+    string(APPEND DEPS_INCLUDE_FLAGS " -I\"${LUAJIT_INCLUDE_DIR}\"")
   elseif(LUA_FOUND)
     set(LUA_ENGINE Lua)
-    string(APPEND DEPS_INCLUDE_FLAGS " -I${LUA_INCLUDE_DIR}")
+    string(APPEND DEPS_INCLUDE_FLAGS " -I\"${LUA_INCLUDE_DIR}\"")
   else()
     message(FATAL_ERROR "Could not find system lua or luajit")
   endif()

